### PR TITLE
feat: Support new task attributes of map(RWA-1447)

### PR DESCRIPTION
**Pull request by dharmendrak**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RWA-1447

### Change description ###

feat: Support new task attributes of map

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by rasikasri — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/wataskmonitor/services/jobs/initiation/InitiationTaskAttributesMapper.java [hunk 1]
```diff
@@ -71,89 +49,38 @@ public List<TaskAttribute> mapTaskAttributes(CamundaTask camundaTask, Map<String
...
+        switch (variable.getType()) {
+
+            case "String":
+                return getVariableValue(variable, String.class, null);
+            case "Boolean":
+                return getVariableValue(variable, Boolean.class, null);
+            case "Integer":
+                return getVariableValue(variable, Integer.class, null);
+            default:
+                return variable.getValue();
```
Comment (BenNzewi) [inline src/main/java/uk/gov/hmcts/reform/wataskmonitor/services/jobs/initiation/InitiationTaskAttributesMapper.java:82]: unit test coverage?

File: src/main/java/uk/gov/hmcts/reform/wataskmonitor/services/jobs/initiation/InitiationTaskAttributesMapper.java [hunk 2]
```diff
@@ -71,89 +49,38 @@ public List<TaskAttribute> mapTaskAttributes(CamundaTask camundaTask, Map<String
...
-            }
+    private Object getCamundaVariableValue(CamundaVariable variable) {
+        switch (variable.getType()) {
+
+            case "String":
+                return getVariableValue(variable, String.class, null);
+            case "Boolean":
+                return getVariableValue(variable, Boolean.class, null);
+            case "Integer":
+                return getVariableValue(variable, Integer.class, null);
```
Comment (BenNzewi) [inline src/main/java/uk/gov/hmcts/reform/wataskmonitor/services/jobs/initiation/InitiationTaskAttributesMapper.java:80]: unit test coverage?

File: src/main/java/uk/gov/hmcts/reform/wataskmonitor/services/jobs/initiation/InitiationTaskAttributesMapper.java [hunk 3]
```diff
@@ -71,89 +49,38 @@ public List<TaskAttribute> mapTaskAttributes(CamundaTask camundaTask, Map<String
...
-            if (value.isPresent()) {
-                return value.get();
-            } else {
-                log.error("could not be mapped");
-                throw new IllegalStateException(
-                    "taskState " + taskStateValue + " could not be mapped to CFTTaskState enum"
-                );
-            }
+    private Object getCamundaVariableValue(CamundaVariable variable) {
+        switch (variable.getType()) {
```
Comment (BenNzewi) [inline src/main/java/uk/gov/hmcts/reform/wataskmonitor/services/jobs/initiation/InitiationTaskAttributesMapper.java:73]: unit test coverage?

File: src/main/java/uk/gov/hmcts/reform/wataskmonitor/services/jobs/initiation/InitiationTaskAttributesMapper.java [hunk 4]
```diff
@@ -71,89 +49,38 @@ public List<TaskAttribute> mapTaskAttributes(CamundaTask camundaTask, Map<String
...
+            attributes.put(ASSIGNEE.value(), camundaTask.getAssignee());
+        }
+        if (camundaTask.getDescription() != null) {
+            attributes.put(DESCRIPTION.value(), camundaTask.getDescription());
+        }
+        attributes.put(TASK_NAME.value(), camundaTask.getName());
+        variables.entrySet().stream()
+            .filter(variable -> !variable.getKey().equals(DUE_DATE.value()))
+            .filter(variable -> !variable.getKey().equals(ASSIGNEE.value()))
+            .filter(variable -> !variable.getKey().equals(DESCRIPTION.value()))
```
Comment (BenNzewi) [inline src/main/java/uk/gov/hmcts/reform/wataskmonitor/services/jobs/initiation/InitiationTaskAttributesMapper.java:66]: unit test coverage? line 66 & 67

File: src/main/java/uk/gov/hmcts/reform/wataskmonitor/services/jobs/initiation/InitiationTaskAttributesMapper.java [hunk 5]
```diff
@@ -71,89 +49,38 @@ public List<TaskAttribute> mapTaskAttributes(CamundaTask camundaTask, Map<String
...
+        attributes.put(CREATED.value(), CAMUNDA_DATA_TIME_FORMATTER.format(camundaTask.getCreated()));
+        if (camundaTask.getAssignee() != null) {
+            attributes.put(ASSIGNEE.value(), camundaTask.getAssignee());
+        }
+        if (camundaTask.getDescription() != null) {
+            attributes.put(DESCRIPTION.value(), camundaTask.getDescription());
+        }
+        attributes.put(TASK_NAME.value(), camundaTask.getName());
+        variables.entrySet().stream()
+            .filter(variable -> !variable.getKey().equals(DUE_DATE.value()))
```
Comment (BenNzewi) [inline src/main/java/uk/gov/hmcts/reform/wataskmonitor/services/jobs/initiation/InitiationTaskAttributesMapper.java:64]: unit test coverage?

File: src/main/java/uk/gov/hmcts/reform/wataskmonitor/services/jobs/initiation/InitiationTaskAttributesMapper.java [hunk 6]
```diff
@@ -71,89 +49,38 @@ public List<TaskAttribute> mapTaskAttributes(CamundaTask camundaTask, Map<String
...
-            new TaskAttribute(TaskAttributeDefinition.TASK_NOTES, null)
-        );
+        Map<String, Object> attributes = new ConcurrentHashMap<>();
+        attributes.put(TASK_TYPE.value(), type);
+        attributes.put(DUE_DATE.value(), CAMUNDA_DATA_TIME_FORMATTER.format(camundaTask.getDue()));
+        attributes.put(CREATED.value(), CAMUNDA_DATA_TIME_FORMATTER.format(camundaTask.getCreated()));
+        if (camundaTask.getAssignee() != null) {
+            attributes.put(ASSIGNEE.value(), camundaTask.getAssignee());
+        }
+        if (camundaTask.getDescription() != null) {
```
Comment (BenNzewi) [inline src/main/java/uk/gov/hmcts/reform/wataskmonitor/services/jobs/initiation/InitiationTaskAttributesMapper.java:59]: unit test coverage?

File: src/main/java/uk/gov/hmcts/reform/wataskmonitor/services/jobs/initiation/InitiationTaskAttributesMapper.java [hunk 7]
```diff
@@ -71,89 +49,38 @@ public List<TaskAttribute> mapTaskAttributes(CamundaTask camundaTask, Map<String
...
-            new TaskAttribute(TaskAttributeDefinition.TASK_REGION_NAME, null),
-            new TaskAttribute(TaskAttributeDefinition.TASK_TERMINATION_REASON, null),
-            new TaskAttribute(TaskAttributeDefinition.TASK_WORK_TYPE, null),
-            new TaskAttribute(TaskAttributeDefinition.TASK_NOTES, null)
-        );
+        Map<String, Object> attributes = new ConcurrentHashMap<>();
+        attributes.put(TASK_TYPE.value(), type);
+        attributes.put(DUE_DATE.value(), CAMUNDA_DATA_TIME_FORMATTER.format(camundaTask.getDue()));
+        attributes.put(CREATED.value(), CAMUNDA_DATA_TIME_FORMATTER.format(camundaTask.getCreated()));
+        if (camundaTask.getAssignee() != null) {
```
Comment (BenNzewi) [inline src/main/java/uk/gov/hmcts/reform/wataskmonitor/services/jobs/initiation/InitiationTaskAttributesMapper.java:56]: unit test coverage?

**Review by BenNzewi — COMMENTED** [review]
Coverage had gone below 80% see coverage comments

**Review by BenNzewi — APPROVED** [review]

**Pull request merged**